src/App.tsx

```diff
@@ -8,53 +8,136 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getData } from './api/fetch';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userId, setUserId] = useState<number | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [isLoading, setIsLoading] = useState(false);
+  const [catchError, setCatchError] = useState(false);
 
-              <Loader />
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedpost] = useState<Post | null>(null);
```

Comment:
The state setter function `setSelectedpost` is inconsistently named. It should be `setSelectedPost` to match the naming convention of `selectedPost` and avoid potential bugs.